Clean up built-in conversions documentation

TWiStErRob opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
 * Instead of duplicating diverging documentation, do a link
   from [`@Property` docs](https://neo4j.com/docs/ogm-manual/current/reference/#reference:annotating-entities:property)
   to [Built-in type conversions](https://neo4j.com/docs/ogm-manual/current/reference/#reference:type-conversion:built-in).
 * List all supported types based on `org.neo4j.ogm.metadata.DomainInfo#registerDefaultFieldConverters()`
 * Update restrictions based on `org.neo4j.ogm.metadata.DomainInfo#set*Converter()`  
   btw, @nmervaillie, what's the reason behind this restriction?  
   https://github.com/neo4j/neo4j-ogm/commit/f3461a3f12905d9c10d30219832f765814df79af#diff-b1f7d7f2edea0677f8f5a32e8ec4af8fR52
 * Update JavaDocs of converters to better reflect what they do, additionally with link to Java 8 time formats.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
No issue link, each of the changes would be a report, I can open an issue (copying the text of this PR) if necessary.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
I refer to the documentation a lot, and I found this section and class JavaDocs to be unsatisfactory for building on it. I was under the impression that **only** `Instant` and `LocalDate` were supported, but then the conversion crashed and I saw in the stack trace how it is created and discovered that other `java.time` classes are also supported which were not mentioned in the docs, but were matching the existing schema I had.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
I tried to do` gradlew html` in the docs folder, but it's using some plugins from a private artifactory repository so failed really fast. I'm also on Windows, so tough luck with the .sh files. Hence couldn't check the links.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document. (**I didn't there's no such file in repo**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed. (**couldn't run docs generation**)

michael-simons commented:
Thanks Rob and sorry for the long wait.

I agree with the JavaDocs 100% and I like the decluttering of the docs (which build just fine btw).

I've fixed the merge conflicts and squashed the commits.

Closed with 1dc8e049e50010dcabc4375849eaec7c698e5979 in master.

TWiStErRob commented:
Wow, I totally forgot about this :) Nice to see it finally in. Thanks.

> I've fixed the merge conflicts and squashed the commits.

Did you know that you can do this on GitHub in browser now?
For trivial conflicts it works nice; and you can edit the PR squash commit message (first line + description). Merge button has a dropdown on the right side, you can configure in Settings which options are enabled.
